### ISSUE TITLE
zapcore: BenchmarkSampler_CheckWithHook consistently fails on Linux with "1094391" is not greater than "1098264"

### ISSUE THREAD

**Issue by odeke-em** [uber-go/zap#996]
Trying to get continuous benchmarking for zap produced consistently failing results per https://dashboard.github.orijtech.com/benchmark/7c6bc13bb23a4c1bbd2f45d6311d0b37 and https://dashboard.github.orijtech.com/benchmark/f6d06d46-d149-41b7-a4b9-b27e24995bb8 and https://dashboard.github.orijtech.com/benchmark/3d30990eb90c440cb26d657bb0323462
<img width="1128" alt="Screen Shot 2021-09-05 at 12 01 59 PM" src="https://user-images.githubusercontent.com/4898263/132138411-9e2f18a9-30a6-4df0-8550-755a2dfb5248.png">

### Continuous benchmarking with gobencher
Given that zap is a critical, high performance logger in shoot outs with other loggers, it helps to have continuous benchmarking. We at Orijtech Inc. have built continuous benchmarking available at the Github marketplace https://github.com/marketplace/gobencher in 1-click, and also the announcement here https://twitter.com/odeke_et/status/1428650768135974919. We have lots of features coming up and we'll make it super simple to track your benchmarks and how they've progressed for every git commit  to this repository **ALL FOR FREE** Our mission is to bring high performance to every developer's doorstep.

**Comment by abhinav** [comment]
Yeah, the benchmark appears to be flaky. We'll try to have someone fix it. 

(Internal issue: GO-858)
